test/Microsoft.AspNet.Razor.Test/Parser/CSharp/CSharpBlockTest.cs

```diff
@@ -736,6 +737,231 @@ public void ParsersCanNestRecursively()
                     Factory.Code("        }").AsStatement().Accepts(AcceptedCharacters.None)));
         }
 
+        public static TheoryData ParseBlockData
+        {
+            get
+            {
+                var factory = CreateDefaultSpanFactory();
+
+                return new TheoryData<string, Block>
+                {
+                    {
+                        "{<span foo='@@' />}",
```

Comment:
Build a helper method member (Func) that builds out the statement block so you don't have to repeat the boiler plate over n over.